manager/src/main/java/com/redhat/service/smartevents/manager/ProcessorServiceImpl.java

```diff
@@ -361,6 +362,19 @@ public Processor updateProcessorStatus(ProcessorManagedResourceStatusUpdateDTO u
                 break;
 
             case UPDATE:
+                // An UPDATE can follow a FAILED_UPDATE when updating a Processor.
+                // If the User has updated a Processor that was previously failed by k8s it has been observed
+                // that the reconciliation loop can first emit an update with the existing FAILED state. This
+                // can lead to the Bridge being FAILED by the BridgeWorker if the Processor is an Error Handler.
+                // Therefore, if a subsequent READY state is emitted we may need to reflect this in the Bridge.
+                if (processor.getType() == ProcessorType.ERROR_HANDLER) {
```

Comment:
Why we want to support this only for error handler processors? Wouldn't this hold for "normal" processors as well?

Reply:
We only propagate Error Handler `Processor` failures to the `Bridge` (otherwise, since they are _internal_, the User is blissfully unaware that something went wrong). If Error Handler `Processor`'s were visible - i.e. listed as a `Processor` in our User facing API - we would not need to propagate the failure.